src/data_readers/data_reader_csv.cpp

```diff
@@ -97,127 +97,157 @@ csv_reader::~csv_reader() {
 }
 
 void csv_reader::load() {
+  int rank = m_comm->get_rank_in_world();
+  MPI_Comm world_comm = m_comm->get_world_comm().comm;
   setup_ifstreams();
   std::ifstream& ifs = *m_ifstreams[0];
-  // TODO: Only one (or a subset) of ranks should read this, then distribute
-  // the results to avoid every rank reading the same file.
-  std::string line;
   // Parse the header to determine how many columns there are.
   // Skip rows if needed.
-  skip_rows(ifs, m_skip_rows);
-  std::streampos header_start = ifs.tellg();
-  // TODO: Skip comment lines.
-  if (std::getline(ifs, line)) {
-    m_num_cols = std::count(line.begin(), line.end(), m_separator) + 1;
-    if (m_skip_cols >= m_num_cols) {
-      throw lbann_exception(
-        "csv_reader: asked to skip more columns than are present");
-    }
-    if (!m_disable_labels) {
-      if (m_label_col < 0) {
-        // Last column becomes the label column.
-        m_label_col = m_num_cols - 1;
-      }
-      if (m_label_col >= m_num_cols) {
+  if (!rank) {
+    skip_rows(ifs, m_skip_rows);
+  } 
+  MPI_Bcast(&m_skip_rows, 1, MPI_INT, 0, world_comm); 
```

Comment:
You should be able to use the existing `lbann_comm::broadcast` method here, which supports scalars.